### PR TITLE
canon-ufrii-driver 10.19.16 (new cask)

### PR DESCRIPTION
**Pull request by unitof**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=pullrequests).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/c/canon-ufrii-driver.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  url "https://downloads.canon.com/bicg2023/drivers/UFRII_v#{version}_mac.zip",
+      verified: "downloads.canon.com/"
+  name "Canon UFR II/UFRII LT/LIPSLX/CARPS2 Printer Driver"
+  desc "Print driver required for Canon imageRUNNER office printers"
+  homepage "https://oip.manual.canon/USRMA-3844-zz-DR-enUV/"
+
+  livecheck do
+    url "https://www.usa.canon.com/bin/canon/support/getsoftwarediver.ds.MACOS_14.39319.All.English.json"
+    regex(/UFRII_V?(\d+(?:\.\d+)+)_Mac/i)
```
Comment (bevanjkay) [inline Casks/c/canon-ufrii-driver.rb:13]:
```suggestion
    regex(/UFRII[._-]v?(\d+(?:\.\d+)+)[._-]Mac/i)
````

File: Casks/c/canon-ufrii-driver.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    strategy :page_match
```
Comment (bevanjkay) [inline Casks/c/canon-ufrii-driver.rb:14]:
```suggestion
```

File: Casks/c/canon-ufrii-driver.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+cask "canon-ufrii-driver" do
+  version "10.19.16"
+  sha256 "58831c5834b8b799b2496848b6c9f4f39d51abd813ad19c7aab38fb046c14be5"
+
+  url "https://downloads.canon.com/bicg2023/drivers/UFRII_v#{version}_mac.zip",
+      verified: "downloads.canon.com/"
+  name "Canon UFR II/UFRII LT/LIPSLX/CARPS2 Printer Driver"
+  desc "Print driver required for Canon imageRUNNER office printers"
```
Comment (bevanjkay) [inline Casks/c/canon-ufrii-driver.rb:8]:
```suggestion
  desc "Printer driver for Canon imageRUNNER office printers"
```

**Review by bevanjkay — COMMENTED** [review]
Please try to add a [zap stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap). The linked documentation provides some recommended methods to create it.

**Comment by unitof** [comment]
I used [Pacifist](https://www.charlessoft.com) and couldn't find anything not already removed by `uninstall` stanza. Added a no-zap-required comment as suggested by nrlquaker/homebrew-createzap.

**Review by bevanjkay — APPROVED** [review]

**Comment by bevanjkay** [comment]
Thank you @unitof

**Pull request merged**